sympy/combinatorics/rewritingsystem.py

```diff
@@ -285,3 +291,111 @@ def reduce(self, word, exclude=None):
                 if new != prev:
                     again = True
         return new
+
+    def construct_automaton(self):
+        '''
+        Construct the automaton based on the set of reduction rules of the system.
+
+        Automata Design:
+        1. The accept states of the automaton are the proper prefixes of the left hand side of the rules.
+        2. The complete left hand side of the rules are the dead states of the automaton
+
+        '''
+
+        generators = [x for x in self.alphabet]
```

Comment:
`list(self.alphabet)` would probably be more efficient.

Reply:
Okay, I've used this in other places too. Will make the changes